cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1119,41 +1119,67 @@ def _get_raw_x_loc(self) -> str:
         else:
             return "X"
 
-    def _is_raw(self, force: bool = False) -> bool:
+    def _has_valid_raw(self, force: bool = False) -> bool:
         """
         Checks if the non-zero values for the raw matrix (adata.X or adata.raw.X)
-        are integers. Returns False if at least one value is not an integer,
+        are positive integers stored as numpy.float32. Also validates that every row contains at least one non-zero
+        value.
 
-        True otherwise.
+        Returns False if at least one value / row does not meet requirements. True otherwise.
 
         Since this process is memory intensive, it will return a cache value if this function has been called before.
         If calculation needs to be repeated use `force = True`
 
         :rtype bool
-        :return False if at least one value is not an integer, True otherwise
         """
         if force:
             self._raw_layer_exists = None
 
         if self._raw_layer_exists is None:
             # Get potential raw_X
-            raw_loc = self._get_raw_x_loc()
-            x = self.adata.raw.X if raw_loc == "raw.X" else self.adata.X
+            x = self.adata.raw.X if self._get_raw_x_loc() == "raw.X" else self.adata.X
```

Comment:
Should use `.get_raw_x()` instead of running logic here with `_get_raw_x_loc()`. It already does this for you.